src/Services/Importers/Finalize.php

```diff
@@ -0,0 +1,52 @@
+<?php
```

Comment:
wrap this service in a job, and dispatch the job in batch->then()

The job will use `SerializesModels` so the dataImport->refresh is not longer needed